addons/account/controllers/download_docs.py

```diff
@@ -1,28 +1,65 @@
+import io
+import zipfile
 
 from odoo import http, _
 from odoo.http import request, content_disposition
 
 
-def _get_zip_headers(content, filename):
+def _get_headers(filename, filetype, content):
     return [
-        ('Content-Type', 'zip'),
-        ('X-Content-Type-Options', 'nosniff'),
+        ('Content-Type', filetype),
         ('Content-Length', len(content)),
         ('Content-Disposition', content_disposition(filename)),
+        ('X-Content-Type-Options', 'nosniff'),
     ]
 
 
+def _build_zip_from_data(docs_data):
+    buffer = io.BytesIO()
+    with zipfile.ZipFile(buffer, 'w', compression=zipfile.ZIP_DEFLATED) as zipfile_obj:
+        for doc_data in docs_data:
+            zipfile_obj.writestr(doc_data['filename'], doc_data['content'])
+    return buffer.getvalue()
+
+
 class AccountDocumentDownloadController(http.Controller):
-    @http.route('/account/export_zip_documents', type='http', auth='user')
-    def export_zip_documents(self, **args):
-        """ Download zipped attachments. """
-        ids = list(map(int, request.httprequest.args.getlist('ids')))
-        filename = request.httprequest.args.get('filename')
-        attachments = request.env['ir.attachment'].browse(ids)
+
+    @http.route('/account/download_invoice_attachments/<models("ir.attachment"):attachments>', type='http', auth='user')
+    def download_invoice_attachments(self, attachments):
         attachments.check_access_rights('read')
         attachments.check_access_rule('read')
-        content = attachments._build_zip_from_attachments()
-        headers = _get_zip_headers(content, filename)
-        return request.make_response(content, headers)
+        assert all(attachment.res_id and attachment.res_model == 'account.move' for attachment in attachments)
+        if len(attachments) == 1:
+            headers = _get_headers(attachments.name, attachments.mimetype, attachments.raw)
+            return request.make_response(attachments.raw, headers)
+        else:
+            inv_ids = attachments.mapped('res_id')
+            if len(set(inv_ids)) == 1:
+                invoice = request.env['account.move'].browse(inv_ids[0])
+                filename = invoice._get_invoice_report_filename(extension='zip')
+            else:
+                filename = _('invoices') + '.zip'
+            content = attachments._build_zip_from_attachments()
+            headers = _get_headers(filename, 'zip', content)
+            return request.make_response(content, headers)
+
+    @http.route('/account/download_invoice_documents/<models("account.move"):invoices>/<string:filetype>', type='http', auth='user')
+    def download_invoice_documents_filetype(self, invoices, filetype, allow_fallback=True):
+        invoices.check_access_rights('read')
+        invoices.check_access_rule('read')
+        docs_data = []
+        for invoice in invoices:
+            get_fn = getattr(invoice, f'_get_invoice_legal_documents_{filetype}', None)
```

Comment:
Please avoid `getattr`, especially like this one where the value comes from a public controller. You can use a common method with `filetype` as argument and each module overrides it to have it’s own logic.

Reply:
@mart-e thank you for the review, I made the requested change.